src/Component/EventDispatcher/src/DependencyInjection/Compiler/RegisterListenerPass.js

```diff
@@ -0,0 +1,102 @@
+        this.dispatcherService = dispatcherService;
+        this.listenerTag = listenerTag;
+        this.subscriberTag = subscriberTag;
+    }
+
+    /**
+     *
+     * @param {Jymfony.Kernel.ContainerBuilder} container
+     */
+    process(container) {
+        if (! container.hasDefinition(this.dispatcherService) && !container.hasAlias(this.dispatcherService)) {
+            return;
+        }
+
+        let definition = container.findDefinition(this.dispatcherService);
+
+        for (let [id, events] of __jymfony.getEntries(container.findTaggedServiceIds(this.listenerTag))) {
+            for (let event of events) {
+                let priority = event['priority'] !== undefined ? event['priority'] : 0;
+
+                if (event['event'] === undefined) {
+                    throw new InvalidArgumentException(__jymfony.sprintf('Service "%s" must define the "event" attribute on "%s" tags.', id, this.listenerTag));
+                }
+
+                if (event['method'] === undefined) {
+                    throw new InvalidArgumentException(__jymfony.sprintf('Service "%s" must define the "method" attribute on "%s" tags.', id, this.listenerTag));
+                }
+
+                definition.addMethodCall('addListenerService', [
+                    event['event'],
+                    [
+                        id,
+                        event['method'],
+                        priority
+                    ]
+                ]);
+            }
+        }
+
+        let extractingDispatcher = new ExtractingEventDispatcher();
+
+        for (let [id, attributes] of __jymfony.getEntries(container.findTaggedServiceIds(this.subscriberTag, true))) {
+            let def = $container.getDefinition(id);
+
+            let myclass = container.getParameterBag().resolveValue(def.getClass());
+            let myInterface = 'Jymfony.EventDispatcher.EventSubscriberInterface';
+
+            if (! myclass.prototype instanceof myInterface) {
+                throw new InvalidArgumentException(__jymfony.sprintf('Service "%s" must implement interface "%s".', id, myInterface));
+            }
+
+            container.addObjectResource(myclass);
+
+            ExtractingEventDispatcher.subscriber = myclass;
+            extractingDispatcher.addSubscriber(extractingDispatcher);
+
+            for (let args of extractingDispatcher.listeners) {
+                args[1] = [new ServiceClosureArgument(new Reference(id), args[1])];
+
+                definition.addMethodCall('addListener', args);
+            }
+
+            extractingDispatcher.listeners = [];
+        }
+    }
+}
+
+class ExtractingEventDispatcher extends implementationOf(EventSubscriberInterface, EventDispatcher) {
+    __constructor() {
+        this._listeners = [];
+    }
+    static get subscriber() {
+        return this._subscriber;
+    }
+
+    static set subscriber(subscriber) {
```

Comment:
Misleading `this` used in static context. If you really want to define a static property, prefer using the class name instead of `this`.